frontend/components/Corpus.js

```diff
@@ -0,0 +1,147 @@
+import React, {useEffect, useState} from "react";
+import * as PropTypes from "prop-types";
+// import STYLES from "./Corpus.module.scss";
+import {getCookie} from "../common";
+import {Modal} from "react-bootstrap";
+
+const Corpus = ({id}) => {
+
+    const [corpusData, setCorpusData] = useState({});
+    const [allDocData, setAllDocData] = useState([]);
+    const [containsDoc, setContainsDoc] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [showModal, setShowModal] = useState(false);
+    const handleOpenModal = () => setShowModal(true);
+    const handleCloseModal = () => setShowModal(false);
+
+    useEffect(() => {
+        fetch(`/api/corpus/${id}`)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+                fetch("/api/all_documents")
+                    .then(response => response.json())
+                    .then(docsData => {
+                        setAllDocData(docsData);
+                        docsData.map((doc) => setContainsDoc((values) => ({
+                            ...values,
+                            [doc.id]: data.documents.includes(doc.id)
+                        })));
```

Comment:
Pedantically, we probably want to `.forEach` here rather than `.map`, since we're not transforming `docsData` for any later purpose.